pkg/ruler/ruler_test.go

```diff
@@ -2004,6 +2005,12 @@ func verifySyncRulesMetric(t *testing.T, reg prometheus.Gatherer, initialCount,
 			cortex_ruler_sync_rules_total{reason="ring-change"} 0
 		`, initialCount, apiChangeCount)), "cortex_ruler_sync_rules_total")
 	})
+	test.Poll(t, time.Second, nil, func() interface{} {
```

Comment:
we have to poll because `sync_rules_total` gets updated at the beginning of the sync and `sync_rules_duration` gets updated at the end of the sync. If this assertion happens in the middle of a sync it'll see mismatched values